subcommands/targets/offline-update.go

```diff
@@ -152,6 +166,31 @@ func getTargetInfo(factory string, targetName string) (*ouTargetInfo, error) {
 	return &info, nil
 }
 
+func isDstDirClean(dstDir string) bool {
+	for _, subDir := range []string{"ostree_repo", "apps"} {
+		fullPath := path.Join(dstDir, subDir)
+		if _, err := os.Stat(fullPath); !os.IsNotExist(err) {
+			fmt.Println(fullPath + " already exists")
+			return false
+		}
+	}
+	return true
```

Comment:
We have the right to require a user providing a clean folder.
So, I'd simply check here if 2 things:
- does the folder exist.
- is it empty.

Let's keep it simple; the user should be the housekeeping.

Reply:
Why not just `does the folder exist.` as currently implemented.

Reply:
Currently we check that `${dir}/apps` and `${dir}/ostree_repo` should not exist.

I propose to only check that `${dir}` either does not exist or is empty; which is a more generic and more strong check.

Reply:
The dir includes `tuf` dir which is OK to be present when the command executed. This is happen when customers want to refresh tuf meta.

Reply:
I prefer to follow the KISS principle here.
I mean, I don't want us to pay attention to _what's OK to have in that folder and what's not OK_.
Even less so I want us to try explaining it to the user.

A requirement to provide an empty folder is justifiable and understandable.
Anything beyond that is convoluted, so I'd only do that if we have a customer request with enough motivation to complicate matters.

Reply:
There is no need to explain anything to a user.
We just discussing here "check $dir" vs "check $dir/app and $dir/ostree_repo".
Is checking two folders compared to checking one folder makes the check so significantly complex? IMO, the code is almost the same.

Reply:
I simply don't get: what is the business use case to check for two folders?

There must be a justifiable answer to that question, _before_ we start talking about the pros and cons.
If there is no customer-facing use case for that, I see no rationale to make our code and support more complex.

Reply:
>I simply don't get: what is the business use case to check for two folders?

There is no need to print any error message and ask a user to remove $dir  if only $dir/tuf  is present. A user does NOT need to know about this detail.

We just simply check if either $dir/apps or $dir/ostree is present. If so the command prints an error and exits. The error message suggest a user to remove the whole `$dst` folder.

Reply:
Also, we should keep in mind that the $dst can be NOT empty, a user may want to to put there some custom data to be transferred to a device. So, in the end, this is up to a user to decide if to delete the whole $dst directory or just $dst/apps and $dst/ostree.

Reply:
> The error message suggest a user to remove the whole $dst folder.

This is where the major part of the confusion comes from.
We should never advice the user to delete `$dst` folder; we should advice to provide an empty folder.
Remember, the `$dst` might be the user's home or the root.

Now, once that confusion is settled, I don't want a user to even know all the details about _which sub-folder is safe to overwrite and which is not_, unless there is a business use case for that. I see no such use case declared in this thread so far.

Reply:
The difference is huge conceptually:
- When I see `foo/bar is not empty, delete this folder`, I naїvely run `rm -rf foo/bar`.
- When I see `foo/bar is not empty, provide an empty folder`, I intuitively run `mkdir -p foo/baz`.

Reply:
>we should advice to provide an empty folder.

This kind of breaks the current API, we've never asked about that, and user expects that they can have anything in that folder.

>Remember, the $dst might be the user's home or the root.

They they may do that. But we need to allow them to have anything they want in $dst folder.

Also, similar argument can be imposed to almost any other `fioctl` command.

Reply:
>When I see foo/bar is not empty, delete this folder, I naїvely run rm -rf foo/bar.

That is why the most safest way to do is to check those two specific folders, and advise to remove the specific folder. If we check just $dst and advise them to delete is then it's not quite save.

Reply:
Hmm... we require a folder to be empty, or the file to not exist _in almost any other fioctl command_ already.

Just IMHO this problem has happened _because_ we failed to require that folder to be empty initially.
So, this seems to be more a bug fix than a breaking change.

Now, if the user wants to put smth into it - they can certainly do that afterwards.
We need not be tied by such restrictive assumptions.

Our job is to perform routine user operations in a safe way.
Requiring that a folder is empty is _safer_ than what you propose @mike-sul.

Reply:
@mike-sul I've got your point.

I'd like to also hear from others on thie subject. @doanac @detsch

Reply:
IMO requiring an empty folder would be sliglthy better, but since there are reasons to not do so, I would keep checking for the 2 specific paths.

About the messaging, I would not suggest any removal, but still present to the user with a hint on what path was detected and eventually led to the command to fail. This is the output from the updated coded:
```
$ fioctl targets offline-update intel-corei7-64-lmp-27 ./offline-update-content --tag main --expires-in-days 15 -f detsch-x86-64 
Checking whether Target exists; target: intel-corei7-64-lmp-27, tag: main, production: false
Refreshing and downloading TUF metadata for Target intel-corei7-64-lmp-27 to offline-update-content/tuf...
Successfully refreshed and downloaded TUF metadata
offline-update-content/ostree_repo already exists
Destination directory already has update data
ERROR: Please provide a clean destination directory or re-run with --allow-multiple-targets to add a new target to a directory which already has update data.
Notice that multiple targets in the same directory is only supported in LmP >= v92.
```

A user struggling with the command from some reason will notice the 
`offline-update-content/ostree_repo already exists` message.

Reply:
@detsch this looks way safer... taking my discussion with @mike-sul to negligance.
Might be a silver lining.